repack.sh

```diff
@@ -163,40 +224,68 @@ repackmsi()
 
 repackcab()
 {
-	CAB="${1}"
-	OUT="${2}"
+	local CAB="${1}"
+	local OUT="${2}"
 
 	echo "==> Extracting ${CAB}"
 
 	# Extract the CAB
-	TEMP="$(mktemp -p . -d)"
+	local TEMP="$(mktemp -p . -d)"
 	gcab -C "${TEMP}" -x "${CAB}" > /dev/null
 
+	# Repack all UEF capsule updates found in the CAB
+	local inffiles=($(grep -lR 'Firmware_Install,UEFI' "${TEMP}"))
 	# Convert all .inf files to unix format
-	find "${TEMP}" -iname '*.inf' -exec sh -c 'dos2unix "$0" > /dev/null 2>&1' {} \;
+	sed -i 's/\r//' "${inffiles[@]}"
```

Comment:
It still tests for dos2unix above and fails if its not present FYI.

I also think there actually were encoding issues in some of the files that dos2unix managed to fix. But maybe that was just my editor going crazy on carriage returns at the time?

How about testing if dos2unix is available, and falling back to sed if its not?

Reply:
While dos2unix can modify the encoding, the manpage for the Linux version says that the default is "ASCII conversion", which converts only line breaks. (Technically, dos2unix for SunOS/Solaris behaves differently as it converts the encoding to ISO 8859-1 --- which is unlikely to help anything these days). 

I suggest removing dos2unix completely. Let me know if that's okay and I'll make the change.

Reply:
Actually, scratch that. It also converts UTF-16 to UTF-8, so it is definitely necessary. I'll put it back in.

Reply:
Okay, I've put dos2unix back into the script (albeit in a tad nicer syntax).